lib/dialects/oracledb/transaction.js

```diff
@@ -71,28 +71,12 @@ module.exports = class Oracle_Transaction extends Transaction {
   }
 
   async acquireConnection(config, cb) {
-    const configConnection = config && config.connection;
-
-    const connection =
-      configConnection || (await this.client.acquireConnection());
-    try {
-      connection.__knexTxId = this.txid;
-      connection.isTransaction = true;
-      return await cb(connection);
-    } finally {
-      debugTx('%s: releasing connection', this.txid);
-      connection.isTransaction = false;
+    return super.acquireConnection(config, async (connection) => {
       try {
-        await connection.commitAsync();
-      } catch (err) {
-        this._rejecter(err);
+        return await cb(connection);
       } finally {
-        if (!configConnection) {
-          await this.client.releaseConnection(connection);
-        } else {
-          debugTx('%s: not releasing external connection', this.txid);
-        }
+        await connection.commitAsync().catch((err) => void this._rejecter(err));
       }
-    }
+    });
```

Comment:
This code was very confusing to follow and mostly duplicated the behavior in the parent class. This simplification apperas to pass the tests that are relevant, but the integration tests are, in general, broken on Oracle. 

I expect to fix those tests up in a separate effort, and certainly don't mind blocking this PR on that work being done.